internal/provider/app_entitlement_resource.go

```diff
@@ -413,6 +426,31 @@ func (r *AppEntitlementResource) Update(ctx context.Context, req resource.Update
 		appEntitlement.ProvisionPolicy = currentAppEntitlement.ProvisionPolicy
 	}
 
+	// TODO(mstanbCO): Need a better pattern for handling this in the merge step, instead of doing this.
+	if currentAppEntitlement.ProvisionPolicy != nil {
+		if currentAppEntitlement.ProvisionPolicy.ConnectorProvision != nil {
```

Comment:
Also on line 414 (I know you didnt change that in this PR) but the if statement doesnt need to be nested, can just be one and statement